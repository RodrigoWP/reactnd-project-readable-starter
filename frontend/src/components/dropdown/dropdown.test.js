'use strict'

import React from 'react'
import { shallow, mount } from 'enzyme'
import { Dropdown } from './index'

const { DropdownItem } = Dropdown

it('Should render correctly', () => {
  const component = (
    <Dropdown children='' />
  )
  expect(shallow(component)).toMatchSnapshot()
})

it('Should render DropdownItem correctly', () => {
  const component = (
    <DropdownItem label='' />
  )
  expect(shallow(component)).toMatchSnapshot()
})

it('Should call onClick when to click in the item', () => {
  const onClick = jest.fn()

  const component = mount(
    <DropdownItem
      label=''
      onClick={onClick}
    />
  )

  component.find('li').simulate('click')

  expect(onClick).toHaveBeenCalledTimes(1)
})

it('When prop icon is not null should render img element', () => {
  const component = mount(
    <DropdownItem label='' icon='icon' />
  )

  const imgElement = component.find('img')

  expect(imgElement.length).toBe(1)
})
