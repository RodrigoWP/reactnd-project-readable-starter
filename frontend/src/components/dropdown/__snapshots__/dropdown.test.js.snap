// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render DropdownItem correctly 1`] = `
<li>
  <span />
</li>
`;

exports[`Should render correctly 1`] = `
<div>
  <img
    src={Object {}}
  />
  <div
    className=""
  />
</div>
`;
